application/dash_application/views/call_ready_ts.py

```diff
@@ -0,0 +1,579 @@
+    # Tables
+    'failed-samples',
+    'data-table'
+])
+
+BAMQC_COL = gsiqcetl.column.BamQc3MergedColumn
+HSMETRICS_COL = gsiqcetl.column.HsMetricsColumn
+ICHOR_COL = gsiqcetl.column.IchorCnaMergedColumn
+CALL_COL = gsiqcetl.column.MutetctCallabilityColumn
+PINERY_COL = pinery.column.SampleProvenanceColumn
+
+
+def dataversion():
+    return DATAVERSION
+
+
+# todo: come back and include whatever else should be a special column
+special_cols = {
+    "Total Reads (Passed Filter)": "Total reads passed filter",
+    "Percent Unique Reads (PF)": "Percent unique reads",
+    "Callability (14x/8x)": "callability",
+    "Purity": "Purity",
+    "File SWID ichorCNA": "File SWID ichorCNA",
+    "File SWID MutectCallability": "File SWID MutectCallability"
+}
+
+ts_table_columns = [*BAMQC_COL.values(), *HSMETRICS_COL.values(), *ICHOR_COL.values(), *CALL_COL.values(),
+                    *special_cols.values()]
+
+
+def get_merged_ts_data():
+    """"
+    Come back and fill in docstring
+    """
+
+    # Sample metadata from Pinery
+    pinery_samples = util.get_pinery_merged_samples(False)  # todo: come back
+    pinery_samples = util.filter_by_library_design(pinery_samples, util.ex_lib_designs)
+
+    ichorcna_df = util.get_ichorcna_merged()
+    ichorcna_df = util.filter_by_library_design(ichorcna_df, util.ex_lib_designs, ICHOR_COL.LibraryDesign)
+
+    hsmetrics_df = util.get_hsmetrics_merged()
+    hsmetrics_df = util.filter_by_library_design(hsmetrics_df, util.ex_lib_designs, HSMETRICS_COL.LibraryDesign)
+
+    callability_df = util.get_mutect_callability()
+    callability_df = util.filter_by_library_design(callability_df, util.ex_lib_designs, CALL_COL.LibraryDesign)
+
+    bamqc3_df = util.get_bamqc3_merged()
+    bamqc3_df = util.filter_by_library_design(bamqc3_df, util.ex_lib_designs, BAMQC_COL.LibraryDesign)
+
+    bamqc3_df[special_cols["Total Reads (Passed Filter)"]] = round(
+        bamqc3_df[BAMQC_COL.TotalReads] / 1e6, 3)
+    ichorcna_df[special_cols["Purity"]] = round(
+        ichorcna_df[ICHOR_COL.TumorFraction] * 100.0, 3)
+    callability_df[special_cols["Callability (14x/8x)"]] = round(
+        callability_df[CALL_COL.Callability] * 100.0, 3)
+
+    ichorcna_df.rename(columns={ICHOR_COL.FileSWID: special_cols["File SWID ichorCNA"]}, inplace=True)
+    callability_df.rename(columns={CALL_COL.FileSWID: special_cols["File SWID MutectCallability"]}, inplace=True)
+
+    # Join IchorCNA and HSMetrics Data
+    ts_df = ichorcna_df.merge(
+        hsmetrics_df,
+        how="outer",
+        left_on=util.ichorcna_merged_columns,
+        right_on=util.hsmetrics_merged_columns)
+
+    # Join IchorCNA+HsMetrics and Callability data
+    ts_df = ts_df.merge(
+        callability_df,
+        how="outer",
+        left_on=util.ichorcna_merged_columns,
+        right_on=util.callability_merged_columns)
+
+    # Join QC data and Pinery data
+    ts_df = util.df_with_pinery_samples_merged(ts_df, pinery_samples, util.ichorcna_merged_columns)
```

Comment:
This join needs to happen after the BamQC3 merge (because otherwise it'll drop all Pinery samples which have no hsmetrics/ichorcna/callability data, but those samples might have bamqc3 data -- this might be why you're not seeing any BamQC metrics)